packages/manager/src/queries/cloudpulse/queries.ts

```diff
@@ -22,10 +23,20 @@ import type {
 const key = 'Clousepulse';
 
 export const queryFactory = createQueryKeys(key, {
-  alerts: {
+  alert: {
+    // contextQueries: {
     // This query key is a placeholder , it will be updated once the relevant queries are added
     queryKey: null,
   },
+  alerts: {
+    contextQueries: {
+      all: (params: Params = {}, filter: Filter = {}) => ({
+        queryFn: () => getAllAlertsRequest(params, filter),
+        queryKey: [params, filter],
+      }),
+    },
+    queryKey: null,
+  },
```

Comment:
Optional, but we could organize this a bit better by moving `alert` inside the context of `alerts`

```suggestion
  alerts: {
    contextQueries: {
      alert: {
        // contextQueries: {
        // This query key is a placeholder , it will be updated once the relevant queries are added
        queryKey: null,
      },
      all: (params: Params = {}, filter: Filter = {}) => ({
        queryFn: () => getAllAlertsRequest(params, filter),
        queryKey: [params, filter],
      }),
    },
    queryKey: null,
  },
```

Reply:
In the future it might be a single `getAlertById` endpoint. Does it make sense for it to be under `alerts._ctx.alert` ?

Reply:
Take this example: You need to invalidate **all** alerts. 

If we structure our query keys like this:
```
["alerts", "all"]
["alert", 1]
```

You have to make multiple invalidations calls to invalidate all alerts:
```typescript
queryClient.invalidateQueries(["alerts"])
queryClient.invalidateQueries(["alert"])
```

<hr />

On the other hand, we structure our query keys like this:
```
["alerts", "all"]
["alerts", "alert", 1]
```

You now can use one invalidation to invalidate **all** alert queries:
```typescript
queryClient.invalidateQueries(["alerts"])
```

React Query matches query keys **based on their prefix** so it is in our best interest to structure our query keys accordingly.

Reply:
Okay will make the suggested changes